File: java/src/MajorityElement.java
import  java.util.Map;
import java.util.HashMap;
public class MajorityElement {
    public int majorityElement(int[] nums) {
        int ans = 0;
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            map.put(nums[i], map.getOrDefault(nums[i], 0) + 1);
        }
        for (int i = 0; i < nums.length; i++) {
            if (map.get(nums[i]) > nums.length/2)
                ans = nums[i];
        }
        return ans;

    }
}

File: python/MajorityElement.py
from typing import List
class MajorityElement:
    def majorityElement(self, nums: List[int]) -> int:
        m = dict()
        for n in nums:
            if n not in m: 
                m[n] = 1
            else:
                m[n] += 1
            if m[n] > len(nums)/2:
                return n
        return 0